bindings/python/edlib.pyx

```diff
@@ -3,10 +3,56 @@ from cpython.mem cimport PyMem_Malloc, PyMem_Free
 
 cimport cedlib
 
+def map_to_bytes(query, target, additional_equalities):
+    cdef bytes query_bytes
+    cdef bytes target_bytes
+    query_needs_mapping = False
+    target_needs_mapping = False
+    if isinstance(query, bytes):
+        query_bytes = query
+    elif isinstance(query, str):
+        query_bytes = query.encode('utf-8')
+        query_needs_mapping = len(query_bytes) != len(query)
+    else:
+        query_needs_mapping = True
+
+    if isinstance(target, bytes):
+        target_bytes = target
+    elif isinstance(target, str):
+        target_bytes = target.encode('utf-8')
+        target_needs_mapping = len(target_bytes) != len(target)
+    else:
+        target_needs_mapping = True
+
+    # Map non-ascii symbols into an ASCII alphabet so it can be used
+    # in the C++ code
+    if query_needs_mapping or target_needs_mapping:
+        query_vals = set(query)
+        target_vals = set(target)
+        input_mapping = {
+            c: chr(idx)
+            for idx, c in enumerate(query_vals.union(target_vals))
+        }
+        if len(input_mapping) > 256:
+            raise ValueError(
+                "query and target combined have more than 256 unique values, "
+                "this is not supported.")
+        query_bytes = ''.join(input_mapping[c] for c in query).encode('ascii')
```

Comment:
Code duplication here, consider creating a lambda function (you can save it to variable) and use that?